cvat/apps/engine/models.py

```diff
@@ -436,6 +436,30 @@ def clear_annotations_in_jobs(job_ids):
         LabeledImageAttributeVal.objects.filter(image__job_id__in=job_ids_chunk).delete()
         LabeledImage.objects.filter(job_id__in=job_ids_chunk).delete()
 
+@transaction.atomic(savepoint=False)
+def clear_annotations_on_frames_in_honeypot_task(db_task: Task, frames: Sequence[int]):
+    if db_task.data.validation_mode != ValidationMode.GT_POOL:
+        # Tracks are prohibited in honeypot tasks
+        raise AssertionError
+
+    for frames_batch in take_by(frames, chunk_size=1000):
+        LabeledShapeAttributeVal.objects.filter(
+            shape__job_id__segment__task_id=db_task.id,
+            shape__frame__in=frames_batch,
+        ).delete()
+        LabeledShape.objects.filter(
+            job_id__segment__task_id=db_task.id,
+            frame__in=frames_batch,
+        ).delete()
+        LabeledImageAttributeVal.objects.filter(
+            image__job_id__segment__task_id=db_task.id,
+            image__frame__in=frames_batch,
+        ).delete()
+        LabeledImage.objects.filter(
+            job_id__segment__task_id=db_task.id,
+            frame__in=frames_batch,
+        ).delete()
```

Comment:
Do you know any other use cases for such a function? Basically, this function is optimized for this use case. A more generic function has to include track handling, and tracks require special processing.

Reply:
> Do you know any other use cases for such a function?

Probably not right now, but it seems that it can be useful when deleting annotations on some specific task frames. 

>  A more generic function has to include track handling, and tracks require special processing.

In that case, you can raise NotImplementedError when deleting tracks. My point is that this function does not really have some specific logic related only to deleting annotations on honeypot frames.

Reply:
It has the custom logic in the lines 441-443. It raises an `AssertionError` if the task is without honeypots, as this deletion is not correct in tasks without honeypots. How raising a `NotImplementedError` is better than `AssertionError`?

Reply:
discussed internally